### PR TITLE
added new validate auth lambda

### PR DESCRIPTION
**Pull request by tjfarns**

### CODE REVIEW

File: backend/src/lambda/validate-auth.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const requestParameters = {
+  id: String,
+  auth: String
+}
+
+async function handler(event: APIGatewayEvent, context: Context) {
+  const { id, auth } = validateRequestBody(event.body, requestParameters)
+
+  const userid = await authenticate(auth)
+  if (userid === id)
```
Comment (nolanbarry) [inline backend/src/lambda/validate-auth.ts:23]: should this be `!==`?

File: backend/src/lambda/validate-auth.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/* 
+ * Users types in OTP and it is submitted to server. Server finds OTP in Security table, 
+ * adds Auth Token to table entry, and returns it to user.
+ */
```
Comment (nolanbarry) [inline backend/src/lambda/validate-auth.ts:4]: I think you forgot to change this?

File: backend/src/lambda/validate-auth.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { authenticate } from '../services/dynamodb'
+import { ExpectedError } from '../model/error'
+import constants from '../utils/constants'
+
+/* See https://docs.aws.amazon.com/lambda/latest/dg/typescript-handler.html */
+
+const requestParameters = {
+  id: String,
+  auth: String
+}
```
Comment (nolanbarry) [inline backend/src/lambda/validate-auth.ts:17]: This is not a problem but a genuine question: do we need the frontend to pass an id? Or should we just return true if the authtoken is valid for any id? Or perhaps return the id that the authtoken is valid for?

File: frontend/src/services/Network.ts [hunk 4]
```diff
@@ -60,6 +60,23 @@ export async function submitOtp() {
     }
 }
 
+export async function validateAuth() {
```
Comment (nolanbarry) [inline frontend/src/services/Network.ts:63]: nice of you to add a sample function to frontend!

File: cdk/lib/lambdas.ts [hunk 5]
```diff
@@ -62,6 +63,10 @@ const configurationProps: { [Property in LambdaReference]: LambdaConfigurationPr
     name: 'login',
     apiRoute: '/user/login'
   },
+  validateAuth: {
+    name: 'validate-auth',
+    apiRoute: '/user/validateauth'
+  },
   getUser: {
     name: 'get-user',
     apiRoute: '/user'
```
Comment (nolanbarry) [inline cdk/lib/lambdas.ts:72]: CDK changes look great!

File: backend/src/lambda/validate-auth.ts (hunk 3 above)
Comment (tjfarns) [reply backend/src/lambda/validate-auth.ts:17]: I think we could get rid of it! I was copying from the format of the otp api endpoint, which verifies against the user in addition to checking for the existence of the otp. It's an added precaution but probably not necessary for something like an authtoken which will likely be longer/more unique than an otp. I'll go ahead and remove it.

**Review by nolanbarry — APPROVED** [review]
great work

**Pull request merged**